SAP Document Management Service Plugin Documentation (#1034)
cap-js/docs: plugins/index.md

[![Node.js logo](../assets/logos/nodejs.svg){style="height:2.5em; display:inline; margin:0 0.2em;"}](https://github.com/cap-js/cap-operator-plugin#readme)
![Java logo](../assets/logos/java.svg){style="height:3em; display:inline; margin:0 0.2em;"}

## SAP Document Management Service {#@cap-js/sdm}

The @cap-js/sdm package is cds-plugin that provides an easy CAP-level integration with SAP Document Management Service. This package supports handling of attachments(documents) by using an aspect Attachments in SAP Document Management Internal Repository.

```cds
extend my.Incidents with {
  attachments: Composition of many Attachments
}
```

![Screenshot showing the Attachments Table in a fiori app](assets/index/sdm-table.png)

> It also provides a CAP-level, easy-to-use integration of the [SAP Document Management Service](https://help.sap.com/docs/document-management-service).

Reviewer: ```suggestion
```

Features:

- Pre-defined type `Attachment` to use in entity definitions

> - Automatic handling of all upload,download,delete and rename operations

Reviewer: ```suggestion
- Automatic handling of all operations including upload, download, delete, and rename
```

> - SAP Document Management Service has inbuilt virus scanning feature.

Reviewer: ```suggestion
- SAP Document Management Service has a built in virus scanning feature.
```

- (Automatic) Fiori Annotations for Upload Controls
- Streaming and piping to avoid memory overloads

> Outlook:

Reviewer: ```suggestion
Future enhancements:
```

Author: is it mandatory to add enhancements or can be skipped

> - Multitenancy intrinsically handled by the plugin

Reviewer: ```suggestion
- Intrinsic multitenancy handling
```
